wallet-new/src/Cardano/Wallet/Kernel/DB/Spec/Update.hs

```diff
@@ -350,5 +350,14 @@ updateUtxo PrefilteredBlock{..} (utxo, balance) =
 updatePending :: PrefilteredBlock -> Pending -> (Pending, Set Txp.TxId)
 updatePending PrefilteredBlock{..} = Pending.removeInputs pfbInputs
 
+-- | Update the foreign transactions with the given prefiltered block
+--
+-- Returns the set of transactions that got removed from the pending set.
+-- Note that this is slightly different from pending transactions as it doesn't
+-- filter on the "inputs" (which aren't part of the wallet for Foreign
+-- transactions) but filter out based on the outputs.
```

Comment:
That is true indeed. I am only filtering on outputs here because the relevant information isn't available in the inputs :/ ... 
Agreed for the ticket :+1: